✨ feat: implement image gallery functionality

de-vasta opened the pull request:
[DEMO LINK](https://de-vasta.github.io/js_gallery_DOM/)

de-vasta committed:
✨ feat: implement image gallery functionality

- Add click event listener to gallery list for image selection
- Update large image display when thumbnail is clicked
- Update demo link with actual GitHub username

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1 +1,14 @@
+galleryList.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  if (e.target.nodeName === 'IMG') {
```
This condition only handles the case where the user clicks directly on the `<img>` element. According to the task requirements, the gallery must also update if the user clicks on the `<a>` tag that wraps the image. You need to adjust the logic to handle both scenarios.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great work on using event delegation! That's the correct approach for this problem. However, I'm requesting a change because the functionality is incomplete. The gallery is required to update when a user clicks on either the thumbnail `<img>` or its surrounding `<a>` tag, but your code currently only handles clicks on the image itself. A good way to solve this is to use the `.closest()` method on the `event.target` to find the parent `<a>` element. This will ensure you can grab the `href` attribute no matter which part of the thumbnail is clicked.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
♻️ refactor(gallery): replace if statement with switch for click handling

- Add support for direct link clicks in addition to image clicks
- Improve code structure and readability
- Add error handling for invalid click targets

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1 +1,22 @@
+      break;
+    }
+    default:
+      throw new Error('Not an image or a link under the click!');
```
While this works, throwing an error for a user click isn't a standard practice. If a user clicks on the space between thumbnails (e.g., on the `<li>` element), this will log an error to the console. A better approach is to simply ignore clicks that are not on the intended targets (`<a>` or `<img>`). You could achieve this by removing the `default` case or having it just `return;`.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work! Your solution meets all the requirements, and the gallery functions perfectly when clicking on either the `<img>` or the surrounding `<a>` tag. I am approving this submission.

I have one minor suggestion for a future improvement. In your `switch` statement's `default` case, it's generally better to simply `return;` rather than throwing an error. This prevents an error from appearing in the console if a user clicks on the space between thumbnails, which is a more standard and user-friendly approach. This is just a best-practice tip, as your current code is fully functional.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

de-vasta committed:
🐛 fix(main): remove error throw for unhandled click events
